data/entities.yaml

```diff
@@ -908,6 +908,28 @@
   properties: [portable]
   capabilities: [arith]
 
+- name: ADT calculator
+  display:
+    attr: device
+    char: '±'
+  description:
+  - |
+    A calculator with Advanced Display Technology (an attached
+    typewriter that can print out the results).  For some reason, in
+    addition to the usual arithmetic on numbers, an ADT calculator can
+    also do arithmetic on types!
+  - |
+    For working with sums of types, the ADT calculator provides
+    the injection functions `inl : a -> a + b` and `inr : b -> a + b`, as
+    well as the case analysis function `case : a + b -> (a -> c) -> (b -> c) -> c`.
+  - |
+    For working with products of types, the ADT calculator enables
+    pair syntax `(a, b)` as well as the projection functions
+    `fst : (a,b) -> a` and `snd : (a,b) -> b`.
```

Comment:
```SUGGESTION
    `fst : a * b -> a` and `snd : a * b -> b`.
```
This might need some extra explanation... :thinking:

Reply:
I like the popular explanation that the product/sum also works for the _size_ of the type.

So that `|Bool * Bool|` is 4, `|() + Bool|` is 3, etc.

Maybe a paragraph with such an example would help illuminate the "arithmetic on types" if the player never heard about it.

Reply:
I like the funny tone of your suggested examples, but I am afraid for someone who has never seen this before they don't really explain it enough.  I tried adding something that explains it a bit more straightforwardly (with some key terms that someone can google).

I still think this won't be enough for someone who has never seen this kind of thing before, but I don't think we can stuff much more in the description. It's worth thinking about what kinds of optional tutorial scenarios we could make that would guide people through familiar FP concepts like lambdas, pair and sum types, etc. if they have never seen those things before.